frame/nfts/src/functions.rs

```diff
@@ -130,6 +134,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			);
 			ensure!(collection_details.attributes == witness.attributes, Error::<T, I>::BadWitness);
 
+			for (account, roles) in CollectionRoleOf::<T, I>::drain_prefix(&collection) {
```

Comment:
what if you have multiple accounts with multiple roles, the client of the call wont about to provide all of them with the current Witness data struct

Reply:
> what if you have multiple accounts with multiple roles

we don't support that

Reply:
I refactored that part